src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -131,17 +132,28 @@ function WebGLShadowMap( _renderer, _objects, _capabilities ) {
 
 			}
 
-			if ( shadow.map === null ) {
+			const switchedToVSM = _previousType !== VSMShadowMap && this.type === VSMShadowMap;
+			const switchedFromVSM = _previousType === VSMShadowMap && this.type !== VSMShadowMap;
+
+			if ( shadow.map === null || switchedToVSM || switchedFromVSM ) {
 
 				const pars = ( this.type !== VSMShadowMap ) ? { minFilter: NearestFilter, magFilter: NearestFilter } : {};
 
+				if ( shadow.map !== null ) {
+
+					shadow.map.dispose();
+
+				}
+
 				shadow.map = new WebGLRenderTarget( _shadowMapSize.x, _shadowMapSize.y, pars );
 				shadow.map.texture.name = light.name + '.shadowMap';
 
 				shadow.camera.updateProjectionMatrix();
 
 			}
 
+			_previousType = this.type;
```

Comment:
When the shadow map type changes from/to VSM, all shadow maps (so all render targets) have to be recreated. Currently, this only happens for the first shadow map.

So I think this line has to be placed outside the for loop. I'll fix this on my side.